File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeNodes(ListNode head) {
        ListNode ans=head.next;
        ListNode t=ans;

        while(t != null){
            int sum=0;
            while(t.val != 0){
                sum += t.val;
                t = t.next;
            }
            ans.val=sum;
            t = t.next;

            ans.next = t;
            ans=ans.next;
        }
        return head.next;
    }
}

File: 2299-merge-nodes-in-between-zeros/merge-nodes-in-between-zeros.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeNodes(self, head: Optional[ListNode]) -> Optional[ListNode]:
        ans = head.next
        traverse = ans

        while (traverse != None):
            values=0
            while(traverse.val != 0):
                values+=traverse.val
                traverse = traverse.next
            ans.val = values
            traverse = traverse.next
            ans.next = traverse
            ans = ans.next
        return head.next

        